### PR TITLE
(#1013)(#1016)(#1017)(#1018) Astro Enhancements and README Updates

### LINKED ISSUE #1013: Add hash to CSS and JS files to bust cache on new deploy
## What New Or Updated Would You Like To See?

The Git SHA hash (or some changing numbers) should be added to the end of the CSS and JS files, so that on deploy it will automatically bust the cache and show the new files. 

## Why Is It Needed?

Right now, it can take up to an hour to show new files, which becomes a problem when CSS and JS has changed.

## Additional Context?

n/a

### LINKED ISSUE #1016: Update README and Dev Container to add recommended VS Code Extensions
## What New Or Updated Would You Like To See?

The README needs updated to add the following extensions:

* MDX
* Astro

The Dev Container/Docker should be checked to ensure these are installed as well.

## Why Is It Needed?

These extension enable code highlighting for the .mdx and .astro languages.

## Additional Context?

n/a

### LINKED ISSUE #1017: Update README on building a production version of the website
## What New Or Updated Would You Like To See?

The README should be updated to give instructions on how to do a production build of the website. 

## Why Is It Needed?

 This is useful for finding hidden errors before pushing up a PR. 

## Additional Context?

This can be done by running `yarn build` and then `yarn preview`.

### LINKED ISSUE #1018: Add sitemap-index.xml to <head>
## What New Or Updated Would You Like To See?

With Astro, a sitemap-index.xml is generated with each deploy. We need to add this to the <head> to [allow crawlers to pick it up more easily.](https://arc.net/l/quote/izugmamx).

## Why Is It Needed?

We need to add this to the <head> to [allow crawlers to pick it up more easily.](https://arc.net/l/quote/izugmamx).

## Additional Context?

n/a

### PR DESCRIPTION
**Pull request by st3phhays**
## Description Of Changes

This addresses a number of small enhancements:

* Adds hash to CSS and JS for automatic cache busting
* Adds information to the README on how to build for production
* Adds information to the README for recommended VS Code extensions
* Adds the sitemap.xml to the <head> to allow search engines to find it easier

## Motivation and Context
All these items came out of the initial Astro conversion and training session.

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made
* [x] Minor documentation fix (typos etc.).
* [x] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [ ] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left-hand side)/
* [ ] Menu structure has been updated

## Related Issue
* Fixes #1013
* Fixes #1016
* Fixes #1017
* Fixes #1018
* ENGTASKS-3795

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,24 +51,55 @@ Follow these steps to open the project in a [Dev Container](https://containers.d
...
+
+The following VS Code extensions are recommended to get the best development experience:
+
+* [Astro](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode) - Syntax highlighting for .astro files.
+* [MDX](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx) - Syntax highlighting for .mdx files.
+* [JavaScript and TypeScript Nightly](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) - JavaScript and TypeScript intelliSense.
+* [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) - Highlights syntax errors in .ts and .js files.
+* [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) - Highlights syntax errors in .md and .mdx files.
+* [Trailing Spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) - Highlights trailing spaces and allows you to easily delete them.
+* [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) - Highlights spelling errors and suggests fixes.
```
Comment (gep13) [inline README.md:100]: I don't think that this is linking to the right extension.

File: src/layouts/Layout.astro [hunk 2]
```diff
@@ -34,6 +34,7 @@ const hash = childProcess
         </style>
 		<meta charset="UTF-8" />
 		<meta name="viewport" content="width=device-width" />
+        <link rel="sitemap" href="/sitemap-index.xml" />
```
Comment (gep13) [inline src/layouts/Layout.astro:37]: Is the spacing at the start of this line correct? It doesn't line up with the others that are already here.

**Review by gep13 — CHANGES_REQUESTED** [review]
Just a couple of minor things.

File: README.md (hunk 1 above)
Comment (st3phhays) [reply README.md]: I have fixed this up.

File: src/layouts/Layout.astro (hunk 2 above)
Comment (st3phhays) [reply src/layouts/Layout.astro:37]: Turns out this page had a mixture of spaces and tabs in it. I have added on a maint commit to make it all formatted correctly.

**Review by gep13 — APPROVED** [review]
LGTM!

**Pull request merged**